app/controllers/OverviewPageController.scala

```diff
@@ -63,9 +67,47 @@ class OverviewPageController @Inject()(inYearAction: InYearAction,
     }
   }
 
+  def handleExclusion(taxYear: Int, incomeSourcesModel: IncomeSourcesModel)(implicit user: User[_]): Unit = {
+    val dividendsRemove = incomeSourcesModel.dividends.exists(dividends => dividends.hasNonZeroData)
+    val giftAidRemove = incomeSourcesModel.giftAid.exists(giftAid => giftAid.hasNonZeroData)
+    val interestRemove = incomeSourcesModel.interest.exists(interests => interests.exists(_.hasNonZeroData))
+    val employmentRemove = incomeSourcesModel.employment.nonEmpty
+    val cisRemove = incomeSourcesModel.cis.nonEmpty
+
+    exclusionUserDataRepository.find(taxYear).map {
+      case Right(value) =>
+        val giftAidHash = incomeSourcesModel.giftAid.exists { data =>  value.map(_.exclusionModel.filter(_.journey == GIFT_AID).head)
+          .exists(_.hash.exists(_ != sha256Hash(data.toString)))
+        }
+        val interestHash = incomeSourcesModel.interest.exists { data =>  value.map(_.exclusionModel.filter(_.journey == INTEREST).head)
+          .exists(_.hash.exists(_ != sha256Hash(data.toString)))
+        }
+      var newExclude: Seq[String] = Seq.empty[String]
+      if (dividendsRemove){
+        newExclude = newExclude :+ DIVIDENDS
+      }
+      if (cisRemove){
+        newExclude = newExclude :+ CIS
+      }
+      if (employmentRemove){
+        newExclude = newExclude :+ EMPLOYMENT
+      }
+      if (giftAidRemove || giftAidHash){
+        newExclude = newExclude :+ GIFT_AID
+      }
+      if (interestRemove || interestHash){
+        newExclude = newExclude :+ INTEREST
+      }
+
+      value.map(_.exclusionModel.filter(data => !newExclude.contains(data.journey))).map { newData =>
+        exclusionUserDataRepository.update(ExclusionUserDataModel(user.nino, taxYear,
+            newData))
+      }
+    }
```

Comment:
This could be a Sequence of tuples that contains the Boolean and which journey gets added.

```
Seq(
  (dividendsRemove, DIVIDENDS),
  (cisRemove, CIS),
  ...
)
```

Also remove the VAR. This can be done functionally. Maybe something like this:

```
Seq(
  (dividendsRemove, DIVIDENDS),
  (cisRemove, CIS),
  ...
).flatMap { case (check, value) =>
 if(check) Some(value) else None
}
```